### ISSUE TITLE
rngSetSeed(0) silently aliases to rngSetSeed(1)

### ISSUE THREAD

**Issue by LeoBuron** [es-ude/OnDeviceTraining#96]
**Audit context:** Filed as part of a USERAPI audit of ODT (Pass 1, April 2026).
Line numbers below reference `es-ude/OnDeviceTraining@b0a737a9d1ff71d71187c6600b56ed5eff4eaab0`.

## Summary

`rngSetSeed` silently aliases seed `0` to seed `1` via the expression `rng.state = seed ? seed : 1;`. Users sweeping seeds across `0..N-1` get a silent duplicate run (seeds 0 and 1 produce identical RNG streams) without any indication.

## Severity

| Axis | Value |
|---|---|
| Priority | Low |
| Affected users | Users running seed sweeps that include 0 |
| Failure mode | Silent |
| Blast radius | Reproducibility / experiment design |

## Root cause

`src/src/arithmetic/RNG.c:44` — https://github.com/es-ude/OnDeviceTraining/blob/b0a737a9d1ff71d71187c6600b56ed5eff4eaab0/src/src/arithmetic/RNG.c#L44

```c
rng.state = seed ? seed : 1;
```

Presumably written to avoid a zero-state for a multiplicative RNG (which would degenerate to producing only zeros for some generator families). The patch silently collapses two distinct seed values into one RNG stream without an error or warning.

## Reproducer

Observed during a 20-seed sweep (seeds `0..19`): two of the 20 runs produced bit-identical CSVs. Source inspection of `rngSetSeed` confirmed the aliasing.

Minimal reproducer:

```c
rngSetSeed(0);
float a = randFloat();
rngSetSeed(1);
float b = randFloat();
assert(a == b);  // passes — they are the same stream
```

The base-project's Python harness works around this by always passing `seed + 1` when spawning ODT.

## Impact

- N-seed sweeps including `seed=0` produce `N-1` distinct streams instead of `N`. Statistics derived from such sweeps under-count sample size by one.
- Any user comparing ODT to PyTorch / NumPy / TensorFlow with `seed=0` on both sides sees unexplained RNG divergence without knowing ODT has silently re-seeded to 1.
- Silent — requires either duplicate-run detection in the experiment harness or source inspection to find.

## Recommended fix

**Option A (recommended):** scramble via `splitmix32(seed)` so all 32-bit seed values including 0 produce distinct, non-zero states.

```c
static uint32_t splitmix32(uint32_t s) {
    s += 0x9E3779B9u;
    s = (s ^ (s >> 16)) * 0x85EBCA6Bu;
    s = (s ^ (s >> 13)) * 0xC2B2AE35u;
    return s ^ (s >> 16);
}

rng.state = splitmix32(seed);
// If splitmix32 ever returns 0 (vanishingly unlikely for a 32-bit seed),
// fall back to some fixed non-zero constant.
```

**Option B:** reject `seed == 0` with `PRINT_ERROR` (after the separate audit issue on `PRINT_ERROR` wiring is resolved, so the error would actually be visible). Preserves current semantics for everyone else but turns the silent alias into a loud rejection.

Option A is preferred — it preserves the "any 32-bit seed is valid" surface with no alias.

## Verification

```c
rngSetSeed(0);
float a0 = randFloat();
rngSetSeed(1);
float a1 = randFloat();
assert(a0 != a1);  // passes iff fix is applied
```

Before fix: assertion fails. After fix: passes.

## Related

Cross-references for this audit batch live in the audit tracking index issue. Fixing the separate `PRINT_ERROR` wiring issue is a prerequisite for Option B (but not Option A, which has no diagnostic requirement).

**Comment by LeoBuron** [comment]
Fixed on `develop` in dee2428 (`fix(#96): rngSetSeed(0) no longer aliases to rngSetSeed(1)`). Closing now since the commit is already on `develop`.
